pkg/api-server/inspect_endpoints.go

```diff
@@ -0,0 +1,132 @@
+package api_server
+
+import (
+	"context"
+	"net"
+	"sort"
+
+	"github.com/emicklei/go-restful"
+
+	api_server_types "github.com/kumahq/kuma/pkg/api-server/types"
+	kuma_cp "github.com/kumahq/kuma/pkg/config/app/kuma-cp"
+	"github.com/kumahq/kuma/pkg/core"
+	config_manager "github.com/kumahq/kuma/pkg/core/config/manager"
+	"github.com/kumahq/kuma/pkg/core/datasource"
+	core_mesh "github.com/kumahq/kuma/pkg/core/resources/apis/mesh"
+	core_manager "github.com/kumahq/kuma/pkg/core/resources/manager"
+	core_model "github.com/kumahq/kuma/pkg/core/resources/model"
+	"github.com/kumahq/kuma/pkg/core/resources/store"
+	rest_errors "github.com/kumahq/kuma/pkg/core/rest/errors"
+	core_xds "github.com/kumahq/kuma/pkg/core/xds"
+	xds_context "github.com/kumahq/kuma/pkg/xds/context"
+	"github.com/kumahq/kuma/pkg/xds/envoy"
+	"github.com/kumahq/kuma/pkg/xds/sync"
+	xds_topology "github.com/kumahq/kuma/pkg/xds/topology"
+)
+
+type simpleMatchedPolicyGetter struct {
+	cfg        *kuma_cp.Config
+	resManager core_manager.ResourceManager
+	cfgManager config_manager.ConfigManager
+}
+
+func NewSimpleMatchedPolicyGetter(cfg *kuma_cp.Config, resManager core_manager.ResourceManager, cfgManager config_manager.ConfigManager) core_xds.MatchedPoliciesGetter {
+	return &simpleMatchedPolicyGetter{
+		cfg:        cfg,
+		resManager: resManager,
+		cfgManager: cfgManager,
+	}
+}
+
+type fakeMetadataTracker struct {
+}
+
+func (f fakeMetadataTracker) Metadata(dpKey core_model.ResourceKey) *core_xds.DataplaneMetadata {
+	return nil
+}
+
+func (s *simpleMatchedPolicyGetter) Get(ctx context.Context, dataplaneKey core_model.ResourceKey) (*core_xds.MatchedPolicies, error) {
+	dataplane := core_mesh.NewDataplaneResource()
+	if err := s.resManager.Get(ctx, dataplane, store.GetBy(dataplaneKey)); err != nil {
+		return nil, err
+	}
+
+	mesh := core_mesh.NewMeshResource()
+	if err := s.resManager.Get(ctx, mesh, store.GetByKey(dataplaneKey.Mesh, core_model.NoMesh)); err != nil {
+		return nil, err
+	}
+
+	dataplanes, err := xds_topology.GetDataplanes(core.Log, ctx, s.resManager, net.LookupIP, dataplaneKey.Mesh)
+	if err != nil {
+		return nil, err
+	}
+
+	proxyBuilder := sync.DefaultDataplaneProxyBuilder(s.resManager, s.resManager, net.LookupIP,
+		datasource.NewDataSourceLoader(s.resManager), *s.cfg, s.cfgManager, &fakeMetadataTracker{}, envoy.APIV3)
+	proxy, err := proxyBuilder.Build(core_model.MetaToResourceKey(dataplane.GetMeta()), &xds_context.Context{Mesh: xds_context.MeshContext{
+		Resource:   mesh,
+		Dataplanes: dataplanes,
```

Comment:
I think that Dataplane Proxy Builder should be provided as a dependency and the logic of building `MeshContext` should be shared between actual XDS and this.

With this in mind, we are slightly changing things with MeshContext and ProxyBuilder with @parkanzky so let's leave it as is. We will adjust the code here after our changes.